tests/integration/test_upgrade.py

```diff
@@ -10,48 +10,48 @@
 import requests
 from juju.application import Application
 from juju.model import Model
-from juju.unit import Unit
 from pytest_operator.plugin import OpsTest
 
 import jenkins
 
-from .constants import ALL_PLUGINS
 from .helpers import (
+    gen_git_test_job_xml,
     generate_client_from_application,
     get_model_jenkins_unit_address,
-    install_plugins,
 )
 
 LOGGER = logging.getLogger(__name__)
 JENKINS_APP_NAME = "jenkins-k8s-upgrade"
+JOB_NAME = "test_job"
 
 
-async def test_jenkins_deploy_with_plugins(ops_test: OpsTest, model: Model):
+async def test_jenkins_deploy_with_job(ops_test: OpsTest, model: Model):
     """
     arrange: given a juju model.
-    act: deploy Jenkins, instantiate the Jenkins client and install the plugins.
+
+    act: deploy Jenkins, instantiate the Jenkins client and define a job.
+
     assert: the deployment has no errors.
     """
     application: Application = await model.deploy(
         "jenkins-k8s",
         application_name=JENKINS_APP_NAME,
-        channel="edge",
-        config={"allowed-plugins": ",".join(ALL_PLUGINS)},
+        channel="stable",
     )
     await model.wait_for_idle(status="active", timeout=30 * 60)
-    address = await get_model_jenkins_unit_address(model, JENKINS_APP_NAME)
-    jenkins_unit = application.units[0]
-    unit_web_client = await generate_client_from_application(ops_test, jenkins_unit, address)
-    await install_plugins(unit_web_client, ALL_PLUGINS)
+    unit_web_client = await generate_client_from_application(ops_test, model, application)
+    unit_web_client.client.create_job(JOB_NAME, gen_git_test_job_xml("k8s"))
 
 
-async def test_jenkins_upgrade_check_plugins(
+async def test_jenkins_upgrade_check_job(
```

Comment:
It looks like this test depends on `test_jenkins_deploy_with_job` - perhaps it would be better to make it a fixture? Generally speaking tests should not depend on each other